### PR TITLE
Added Use of Real Browser, Category, Delivery Country, Reviews, Support for UAE

### LINKED ISSUE #11: Add optional Puppeteer
Unless we solve the larger blocking of Cheerio, it would be good to add Puppeteer as an option as maybe even default.

### LINKED ISSUE #13: Add Amazon.ae (United Arab Emirates)
requested by user on intercom

### PR DESCRIPTION
**Pull request by Ferencz**

### CODE REVIEW

**Comment by pocesar** [comment]
hey @Ferencz I'll review your PR shortly

File: INPUT_SCHEMA.json [hunk 1]
```diff
@@ -30,53 +37,612 @@
...
+              "software-intl-ship",
+              "sporting-intl-ship",
+              "tools-intl-ship",
+              "toys-and-games-intl-ship",
+              "videogames-intl-ship"
+          ]
       },
+    "searchType": {"title": "Select the type of search you would like",
+          "type": "string",
+          "description": "Select your Amazon domain",
```
Comment (pocesar) [inline INPUT_SCHEMA.json]: wrong description here

File: src/utils.js [hunk 2]
```diff
@@ -88,6 +88,8 @@ function getCurrency(request) {
             return 'EUR';
         case 'www.amazon.co.jp':
             return 'JPY';
+        case 'www.amazon.ae':
```
Comment (pocesar) [inline src/utils.js:91]: this is missing from the INPUT_SCHEMA

File: INPUT_SCHEMA.json [hunk 3]
```diff
@@ -30,53 +37,612 @@
...
+        "description": "Keywords, asins or directUrls you would like to extract from Amazon, comma separated",
+        "prefill": "Iphone X,Samsung monitor 27 QHD",
+        "editor": "textarea"
+    },
     "maxResults": {
       "title": "Number of results",
       "type": "integer",
       "description": "Number of results you would like to save in total.",
-      "default": 0,
+      "default": 20,
```
Comment (pocesar) [inline INPUT_SCHEMA.json]: also set the `"prefill"`, people might get confused why the request queue is stopping in 60. the original default of 0 meant "infinite"

File: src/createSearchUrls.js [hunk 4]
```diff
@@ -26,10 +31,123 @@ async function createSearchUrls(input) {
...
+    if ((!input.keywords) && (!input.asins) && (!input.directUrls) && (!input.search)) {
         throw new Error('Keywords/Asins required');
     }
+    if (!input.searchType) {
+        throw new Error('SearchType required');
+    }
+    if (input.searchType === 'asins') {
+        try {
+            try {
+                console.log(typeof JSON.parse(input.search))
```
Comment (pocesar) [inline src/createSearchUrls.js]: console statement

File: src/main.js [hunk 5]
```diff
@@ -63,105 +102,86 @@ Apify.main(async () => {
...
-                        const paginationEle = $('ul.a-pagination li.a-last a');
-                        if (paginationEle.length !== 0) {
-                            paginationUrlSeller = urlOrigin + paginationEle.attr('href');
-                        } else {
-                            paginationUrlSeller = false;
+    const pptr = new Apify.PuppeteerCrawler({
+        requestQueue,
+        launchPuppeteerOptions: {
+            headless: true,
+            slowMo: 100,
```
Comment (pocesar) [inline src/main.js]: use `slowMo: Apify.isAtHome() ? 100 : undefined` here

File: src/utils.js (hunk 2 above)
Comment (Ferencz) [reply src/utils.js:91]: Fixed

File: INPUT_SCHEMA.json (hunk 3 above)
Comment (Ferencz) [reply INPUT_SCHEMA.json]: Fixed with a prefill of 20 results

File: src/createSearchUrls.js (hunk 4 above)
Comment (Ferencz) [reply src/createSearchUrls.js]: Removed the log used for debugging

File: src/main.js (hunk 5 above)
Comment (Ferencz) [reply src/main.js]: Implemented the change

**Pull request merged**